### PR TITLE
test: setup unit tests

### PR DESCRIPTION
**Pull request by eliobischof**

### CODE REVIEW

**Comment by peintnermax** [comment]
lets add the output of cypress to `.gitignore`

# cypress
apps/login/cypress/screenshots
apps/login/cypress/videos

**Comment by peintnermax** [comment]
don't we need a `jest-setup.ts` in apps/login for setting up the testing library `import '@testing-library/jest-dom/extend-expect'` just like in zitadel-react ?

File: packages/zitadel-server/package.json [hunk 1]
```diff
@@ -11,21 +11,28 @@
...
     "dev": "tsup --dts --watch",
     "lint": "eslint \"src/**/*.ts*\"",
-    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
-    "prebuild": "pnpm run generate",
-    "generate": "buf generate https://github.com/zitadel/zitadel.git --path ./proto/zitadel"
+    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
   },
   "devDependencies": {
     "@bufbuild/buf": "^1.14.0",
+    "@mocks-server/main": "^4.1.0",
```
Comment (peintnermax) [inline packages/zitadel-server/package.json:25]: do we need this?

**Review by peintnermax — CHANGES_REQUESTED** [review]

**Comment by eliobischof** [comment]
> don't we need a `jest-setup.ts` in apps/login for setting up the testing library `import '@testing-library/jest-dom/extend-expect'` just like in zitadel-react ?

We just need the "@testing-library/jest-dom/extend-expect" module loaded before the test suite. That's with the setupFilesAfterEnv property in jest.config.ts.

I fixed it in zitadel-react too.

**Comment by eliobischof** [comment]
> lets add the output of cypress to `.gitignore`
> 
> ```
> # cypress
> apps/login/cypress/screenshots
> apps/login/cypress/videos
> ```

I removed the whole cypress folder for now.

File: packages/zitadel-server/package.json (hunk 1 above)
Comment (eliobischof) [reply packages/zitadel-server/package.json]: uninstalled it :+1:

File: package.json [hunk 2]
```diff
@@ -1,16 +1,18 @@
 {
   "private": true,
   "scripts": {
+    "generate": "turbo run generate",
+    "prebuild": "turbo run generate",
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply package.json:5]: right, but the turbo.json is project scoped, right? So it still makes sense to keep the prebuild script in the package.json, for cases you just want to build the zitadel-client. What do you think?

The generate task is already defined as a dependency of the build task in turbo.json.

File: packages/zitadel-client/package.json [hunk 3]
```diff
@@ -10,20 +10,27 @@
     "dist/**"
   ],
   "scripts": {
+    "generate": "buf generate https://github.com/zitadel/zitadel.git --path ./proto/zitadel",
+    "prebuild": "pnpm run generate",
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply packages/zitadel-client/package.json:14]: right, but the turbo.json is project scoped, right? So it still makes sense to keep the prebuild script in the package.json, for cases you just want to build the zitadel-client. What do you think?

The generate task is already defined as a dependency of the build task in turbo.json.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (eliobischof) [reply package.json:5]: The generate task is already defined as a dependency of the build task in turbo.json.

I removed it here :+1:

**Review by peintnermax — APPROVED** [review]

**Pull request merged**